advanced/wallets/react-wallet-v2/src/views/SessionProposalModal.tsx

```diff
@@ -216,6 +218,29 @@ export default function SessionProposalModal() {
     }
   }, [])
 
+  // connect potential smart account address to wallet
+  {supportedChains.length &&
+    supportedChains.map((chain, i) => {
+        let enabledChain
+        switch (chain?.chainId) {
+          case 5:
+            enabledChain = 'goerli'
+            break;        
+          case 11155111:
+            enabledChain = 'sepolia'
+            break;
+        }
+
+        const pKey = eip155Wallets[getAddress(chain?.namespace) as `0x${string}`].getPrivateKey() as `0x${string}`
+        const smartAccountClient = new SmartAccountLib(pKey, enabledChain as SmartAccountEnabledChains)
+      
+        smartAccountClient.getSmartAccountAddress().then(res => {
```

Comment:
is there a chance that `getSmartAccountAddress` fails for some reason? 
Simple catch and log of the error might prevent it causing issues with other functionality in the wallet